src/Routing/PushState.purs

```diff
@@ -212,3 +200,38 @@ matchesWith parser cb = foldPaths go (go Nothing)
     maybe (pure a) (\b -> Just b <$ cb a b)
       <<< indexl 0
       <<< parser
+
+-- | Similar to `setImmediate`, it's impelemnted using microtask queue via MutationObserver
+-- | to schedule callbacks. this way it's more imediate then `setTimout` would have been.
+-- | We use a fresh counter so that the text change mutation always fires.
+-- |
+-- | NOTE: it's not that unsafe, but we use `unsafeRunRef` and `runPure` so it's still unsafe.
+unsafeSetImmediate
+  :: forall r
+  . Eff (ref :: REF, dom :: DOM |r) Unit -> Eff (ref :: REF, dom :: DOM |r) Unit
```

Comment:
Why use unsafe and runPure? https://github.com/natefaubion/purescript-spork/blob/master/src/Spork/Scheduler.purs#L20

We are already in Eff, creating a PushStateInterface. There's no reason to have a global `unsafeSetImmediate` created with unsafe effects. the microtask queue is already global.

Reply:
This way one node is used for all calls to `unsafeSetImmediate eff`.  but for this case it's not important